src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/LightCompiler.cs

```diff
@@ -2179,6 +2179,25 @@ private void EmitThisForMethodCall(Expression node)
             CompileAddress(node, -1);
         }
 
+        private static bool ShouldWritebackNode(Expression node)
+        {
+            if (!node.Type.GetTypeInfo().IsValueType)
+                return false;
+            switch (node.NodeType)
+            {
+                case ExpressionType.Parameter:
+                case ExpressionType.Call:
+                case ExpressionType.ArrayIndex:
```

Comment:
No. Though it makes no difference, since the behaviour for `Unbox` is never to write-back anyway in the subsequent code, as after all, unboxing directly to a `ref` would create a new local whose being over-written would have no effect on its source.

Comparable case where `C#` doesn't have the effect write-back emulates:
```C#
object o = new List<int>{1}.GetEnumerator();
Console.WriteLine(((List<int>.Enumerator)o).MoveNext());
Console.WriteLine(((List<int>.Enumerator)o).MoveNext());
```

The compiler matches the IL from that pretty close (`ldloca` and `call`, but against a local used only for that unboxed value), and hence matches the C# behaviour.

The interpreter should therefore also return false here.

(It's also worth noting that the next step means that `Call` only results in a writeback if the call is to `Array.Get()` with its relatively unusual reference-returning behaviour). Returning `true` here means it can be a write-back case if the logic for arguments would have it as such, not that it necessarily will.